src/main/java/org/mule/extension/http/api/error/HttpError.java

```diff
@@ -132,11 +132,25 @@
   }
 
   public static Set<ErrorTypeDefinition> getHttpRequestOperationErrors() {
-    return ImmutableSet.<ErrorTypeDefinition>builder()
-        .add(PARSING).add(TIMEOUT).add(SECURITY).add(CLIENT_SECURITY).add(CONNECTIVITY).add(BAD_REQUEST).add(FORBIDDEN)
-        .add(UNAUTHORIZED).add(METHOD_NOT_ALLOWED).add(TOO_MANY_REQUESTS)
-        .add(NOT_FOUND).add(UNSUPPORTED_MEDIA_TYPE).add(NOT_ACCEPTABLE).add(INTERNAL_SERVER_ERROR).add(SERVICE_UNAVAILABLE)
-        .build();
+    final Set<ErrorTypeDefinition> httpRequestOperationErrors = new HashSet<>();
+
+    httpRequestOperationErrors.add(PARSING);
+    httpRequestOperationErrors.add(TIMEOUT);
+    httpRequestOperationErrors.add(SECURITY);
+    httpRequestOperationErrors.add(CLIENT_SECURITY);
+    httpRequestOperationErrors.add(CONNECTIVITY);
+    httpRequestOperationErrors.add(BAD_REQUEST);
+    httpRequestOperationErrors.add(FORBIDDEN);
+    httpRequestOperationErrors.add(UNAUTHORIZED);
+    httpRequestOperationErrors.add(METHOD_NOT_ALLOWED);
+    httpRequestOperationErrors.add(TOO_MANY_REQUESTS);
+    httpRequestOperationErrors.add(NOT_FOUND);
+    httpRequestOperationErrors.add(UNSUPPORTED_MEDIA_TYPE);
+    httpRequestOperationErrors.add(NOT_ACCEPTABLE);
+    httpRequestOperationErrors.add(INTERNAL_SERVER_ERROR);
+    httpRequestOperationErrors.add(SERVICE_UNAVAILABLE);
+
+    return unmodifiableSet(httpRequestOperationErrors);
```

Comment:
What about make httpRequestOperationErrors a static method variable and creating it when it's null? This way you can avoid repeated additions, hash calculations... I don't know how many times is this method used